/**:
  ros_parameters:
    holoocean_scenario:
      name: hovering_dynamics
      package_name: Ocean   
      world: PierHarbor
      main_agent: auv0      
      ticks_per_sec: 30
      frames_per_sec: 0
      draw_arrow: true
      show_viewport: false
      weather:
        - type: sunny
      agents:
        - agent_name: auv0
          agent_type: CougUV
          control_scheme: 1  
          # location: [227, 63, -285]
          # location: [450, 144, -287] Open Water good spot
          location: [497, -631, -7]
          rotation: [0, 0, 90]
          publish_commands: true
          sensors:
            - sensor_type: DynamicsSensor
              configuration:
                UseCOM: true
                UseRPY: false
              ros_publish: true
            - sensor_type: DynamicsSensor
              sensor_name: Modem
              Hz: 10
              configuration:
                UseCOM: true
                UseRPY: false
              ros_publish: true
            - sensor_type: DVLSensor
              ros_publish: true
              Hz: 12
              configuration:
                VelCov: [0.001, 0.001, 0.001, 0.001]
                MaxRange: 100
            - sensor_type: IMUSensor
              ros_publish: true
              Hz: 10
              # configuration:
              #   AccelCov: [0.001, 0.001, 0.001]
              #   AngVelCov: [0.001, 0.001, 0.001]
            - sensor_type: GPSSensor
              ros_publish: true
              Hz: 1
              configuration:
                # Cov: [0.0001, 0.0001, 0.0001]
                Depth: 10.5
                # DepthCov: 0.05
            # - sensor_type: LocationSensor
            #   ros_publish: true
            - sensor_type: RotationSensor
              ros_publish: true
              Hz: 10
            - sensor_type: DepthSensor
              Hz: 12
              ros_publish: true
              # configuration:
              #   Cov: 0.001
            # - sensor_type: VelocitySensor
            #   ros_publish: true
            # - sensor_type: SidescanSonar
            #   ros_publish: false
            #   conifguration:
            #     ViewRegion: true
            # - sensor_type: RGBCamera
            #   ros_publish: true
            #   sensor_name: "TempViewport"
            #   socket: "CameraSocket"
            #   location: [-0.7, -1.0, -3.0]
            #   rotation: [0.0, -80.0, 0.0]
            #   Hz: 30
            #   configuration:
            #     CaptureWidth: 1200
            #     CaptureHeight: 1200
          # dynamics: 
          #   mass: 16
          #   length: 0.5
          #   rho: 1026
          #   diam: 0.1
          #   r_bg: [0, 0, 0.02]
          #   r_bb: [0, 0, 0]
          #   Cd: 0.42
          #   # K_nomoto:  
          actuator: 
            fin_count: 3
            fin_offset_deg: 270
            fin_area: 0.00665 #0.00665
            deltaMax_fin_deg: 25
            nMax: 1525
            CL_delta: 1.5
          # autopilot: 
          #   depth:
          #     wn_d_z: 0.2
          #     Kp_z: 0.5
          #     T_z: 100
          #     Kp_theta: 4.0
          #     Kd_theta: 2.3 #2.3
          #     Ki_theta: 0.6 #0.6
          #     K_w: 5.0
          #     theta_max_deg: 30
          #   heading:
          #     wn_d: 0.2
          #     zeta_d: 0.8
          #     r_max: 0.9
          #     lam: 0.1
          #     phi_b: 0.1
          #     K_d: 0.0 #0.5
          #     K_sigma: 0.05
          #   surge:
          #     kp_surge: 700.0
          #     ki_surge: 0.0 #100.0
          #     kd_surge: 0.0 #40.0
    
/controller_node:
  ros_parameters:
    torpedo_fin: 3
    vehicle_name: auv0

/command_node:
  ros_parameters:
    random: false
